# rates of timers in Hz
rate:
  check_subscribers: 1 # [Hz]
  check_state_machine: 4 # [Hz]
  check_forbidden: 2
  check_balloons: 20
  check_emulation: 10
  state_machine: 30
  pub_rviz: 30
  pub_status: 15
# defines area for drone to operate
area:
  x_min: -5
  x_max: 5
  y_min: -5
  y_max: 5
  z_min: 0.5
  z_max: 5
  offset: 1.5
# arena coords are like this [x_min,x_max,y_min, y_max, z_min,z_max, offset]
# arena:
#   left: [-20,0,-20,20, 1,8,3]
#   right: [0,20,-20,20, 1,8,3]
#   all: [-20,20,-20,20, 1,8,3]

arena:
  [
  -45,0,-10,10, 0.5,6,0.5,
  0,45,-10,10, 0.5,6,0.5,
  -45,45,-10,10, 0.5,6,0.5,
  ]


safe1: [
11.934764, 12.617803,
6.81, 13.73,
7.43, 18.28,
-27.33, 28.30,
-29.02, 24.79,
-33.31, 25.58,
-39.79, 1.88,
-38.11, 0.02,
-38.08, -2.50,
-3.64, -12.25,
-1.55, -12.32,
2.838657, -14.000979,
]

safe2: [
2.838657, -14.000979,
9.02, -16.39,
47.15, -27.94,
48.90, -27.03,
56.24, -2.26,
53.08, -0.97,
54.58, 4.14,
19.66, 16.03,
17.04, 10.87,
11.934764, 12.617803
]

safe_big: [
-38.08, -2.50,
-3.64, -12.25,
-1.55, -12.32,
9.02, -16.39,
47.15, -27.94,
48.90, -27.03,
56.24, -2.26,
53.08, -0.97,
54.58, 4.14,
19.66, 16.03,
17.04, 10.87,
6.81, 13.73,
7.43, 18.28,
-27.33, 28.30,
-29.02, 24.79,
-33.31, 25.58,
-39.79, 1.88,
-38.11, 0.02
]


constraints:
  sweeping: "sweeping"
  going: "aggressive"
  attack: "attack"
# time thresholds for states - means, if dron stays in area for this amount a time, it'll go to fobidden
time_thresholds:
  going_to: 30
  destroy: 20
  checking_balloon: 30
# should the UAV land after destroying all balloons
land_at_the_end: false
#height offset when killing the balloon
height_offset: 0.3
#height where drone will operate ( up to this height )
elips_height: 4.5
#idle_time
idle_time: 0
#velocity for goTo's
vel: 3
#velocity for attack
vel_attack: 4
#velocity of going around arena
vel_arena: 4
#arena corner velocity factor
arena_corner_factor: 2.5
#distance to the balloon
dist_to_balloon: 5
#distance to kill the balloon
dist_to_overshoot: 4
#overshoot offset
overshoot_offset: 0.15
#trajectory length for MPC
traj_len: 40
#traj time for MPC
traj_time: 8
#distance error to target the closest balloon so, it will be compared to current chosen via KF + dist_error 
dist_error: 3
#dist accuracy to be as close to the balloon
dist_acc: 2.5
# distance to activate KF flying
dist_kf_activation: 5.5
#time for waiting for KF reset
wait_for_ball: 3
#max time balloon not seen in PCL
max_time_balloon: 3
#how many time to reset the KF
reset_tries: -1
#time to land - max time of not observing any balloons at all 
time_to_land: 120 # seconds 
#world frame, all points are tran
# number of trying to hit the same balloon
balloon_tries: 2
#forbidden area radius
forbidden_radius: 4.5
#forbidden time
forbidden_time: 30
# height tolerance of the arena
height_tol: 0.5
# dead band distance factor
dead_band_factor: 0.5
# timer to count the balloon as popped - for emulation
time_to_emulate: 4
# dist to activate flying by PCL z
balloon_activation_dist: 11
# scan step 
arena_scan_step: 7
#is arena swap enabled ? 
is_arena_swap_enabled: false
# time when arena swaps
arena_swap_time: 500
# time wait for comrade
max_silence_time: 30
# if the height is the same
cheating_mode: false
cheating_height: 5

